### PR TITLE
Status change modals

### PR DESCRIPTION
**Pull request by jerrycyip**
### :scroll: Description

- Describe the general shape of this PR (new feature? refactor? bug fix? one-line change?)
Implement a common confirmation modal for confirmation of admin updates (such as updating: meeting status, agenda item status, delete meeting etc.) that is mobile friendly and in line with the latest [MVP figma designs](https://www.figma.com/file/MEBv8hZLr4MICzKCVXADjg/Happening-ATM?type=design&node-id=6150-89385&mode=design&t=nU8nQufh2A1RrPAn-0).  In the process, remove outdated, unused, and duplicative code.

- Describe what changes are being made
Implement a common confirmation modal for confirmation of admin updates (such as updating: meeting status, agenda item status, delete meeting etc.) that is mobile friendly and in line with the latest [MVP figma designs](https://www.figma.com/file/MEBv8hZLr4MICzKCVXADjg/Happening-ATM?type=design&node-id=6150-89385&mode=design&t=nU8nQufh2A1RrPAn-0).  In the process, remove outdated, unused, and duplicative code.

- Describe why these changes are being made
Original code did not match figma designs and was not 'DRY' code (i.e. 'Do-not-repeat-yourself').  There were 3 different confirmation modal components (for confirming an update to meeting status, to agenda item status, or deletion of a meeting) which seemingly had a lot of redundancies and un-necessary complexity.  
There also appeared to be un-necessary use of certain React provided features such as the 'useCallback' hook, which similarly added complexity with no measurable benefit or need to the simple use case of a confirmation modal(perhaps it was implemented at the time for developers to test out such React features?).  Such un-necessary code has been removed.

- List the use cases and edge cases relevant to this PR
new functionality matches latest [MVP figma designs](https://www.figma.com/file/MEBv8hZLr4MICzKCVXADjg/Happening-ATM?type=design&node-id=6150-89385&mode=design&t=nU8nQufh2A1RrPAn-0)

Future note: whereas attempts were made to consolidate as much functionality into this component as possible (notice  10+ deleted files), due to time and complexity constraints, not all modal use-cases were addressed.  In particular, dynamic modals that require forms such as date selectors (e.g. for new meeting creation) or time selectors (e.g. for setting time-specificity to an agenda item) or selecting a file (e.g. for csv agenda uploads) were not encompassed in this effort.  Those features work as is already, so it was decided that it was not a proper use of time and resources to try and fix what isn't currently 'broken'.  A future effort would be necessary to revisit the potential for consolidating such functionality into more re-usable and singular component(s).

- Describe how errors will be handled. How will we know if this code breaks in production
error handling code added to address any potential errors when calling the database via applicable GraphQL mutations for updating the database 

- Describe any external libraries/dependencies added or removed in this PR
N/A - no newly introduced libraries/dependencies.  Re-use of already used [react-modal library](https://reactcommunity.org/react-modal/) to ensure confirmation modals have WAI-ARIA accessibility (aka 'a11y') support.  react-modal is used already for other modals in this application.

- Describe any security risks are there regarding this change
N/A

- Describe how you tested these changes
Tested before and after functionality including desktop and mobile views on latest popular browsers (Chrome, Firefox, Edge, Safari) to ensure no unexpected behavior and confirm functionality matches latest MVP designs.

- Link to relevant external documentation
See latest [MVP figma designs](https://www.figma.com/file/MEBv8hZLr4MICzKCVXADjg/Happening-ATM?type=design&node-id=6150-89385&mode=design&t=nU8nQufh2A1RrPAn-0)

--------------------------
### :clipboard: Mandatory Checklist

- [x] Example of a checked item (please remove when creating your Pull Request) 

- [x] Linked to the Github Issues being addressed using the right sidebar :arrow_right:
- [x] Have you discussed these changes with the project leader(s)?
- [x] Do all variable and function names communicate what they do?
- [x] Were all the changes commented and / or documented?
- [x] Is the PR the right size? (If the PR is too large to review, it might be good to break it up into multiple PRs.)
- [x] Does all work in progress, temporary, or debugger code have a TODO comment with links to Github issues?
- [x] *If you changed the user interface, did you add before and after screenshots to below?*

--------------------------
#### Before

https://github.com/codeforsanjose/project-happening-atm/assets/54079796/d8301891-3d6f-43d9-9281-7e64cb6d3633

#### After

https://github.com/codeforsanjose/project-happening-atm/assets/54079796/9eacc562-73fa-4f31-a3e8-d1e905af7ae6

--------------------------
### :blue_book: Glossary
- PR = Pull Request

### CODE REVIEW

File: frontend/src/components/MeetingView/AgendaView/AgendaItem.jsx [hunk 1]
```diff
@@ -146,51 +147,100 @@ const RenderedAgendaItem = forwardRef(({ subStatus, args, ...props }, ref) => {
...
     {
-      label: 'meeting.tabs.agenda.status.options.in-progress',
+      label: "meeting.tabs.agenda.status.options.in-progress",
       value: MeetingItemStates.IN_PROGRESS,
-      class: 'in-progress',
+      class: "in-progress",
     },
     {
-      label: 'meeting.tabs.agenda.status.options.on-hold',
+      label: "meeting.tabs.agenda.status.options.on-hold",
```
Comment (JMStudiosJoe) [inline frontend/src/components/MeetingView/AgendaView/AgendaItem.jsx:163]: this is a nit pick but since this is not a json file lets use single quotations in jsx files. Typically its easier to use the " when you need them in either HTML tags or in JSON files but lets keep it to ' quotes for javascript logic

File: frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx [hunk 2]
```diff
@@ -1,45 +1,45 @@
-import React, { useEffect, useState } from 'react';
-import { useMutation } from '@apollo/client';
+import React, { useEffect, useState } from "react";
+import { useMutation } from "@apollo/client";
```
Comment (JMStudiosJoe) [inline frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx:2]: same here for quotes, likely a linter fix

File: frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx [hunk 3]
```diff
@@ -153,40 +153,40 @@ const UpdateItemStartTimeModal = ({ args }) => {
...
-              <label className='meridian-btn' htmlFor="am">
+              <label className="meridian-btn" htmlFor="am">
                 <span>AM</span>
               </label>
               <input
                 type="radio"
                 name="meridian"
                 id="pm"
-                value='PM'
+                value="PM"
```
Comment (JMStudiosJoe) [inline frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx:163]: in this part is ok double quotes

Comment (JMStudiosJoe) [reply frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx:163]: but its totally fine if not

File: frontend/src/components/ConfirmationModal/ConfirmationModal.scss [hunk 4]
```diff
@@ -0,0 +1,112 @@
+@import '../../utils/colors';
+
+.confirmation-modal{
+	margin-top: -6rem;
+	width: 23rem;
+	height: 17rem;
+  font-family: "Poppins", sans-serif;
```
Comment (JMStudiosJoe) [inline frontend/src/components/ConfirmationModal/ConfirmationModal.scss:7]: think some spacing issues here too :)

File: frontend/src/components/MeetingView/AgendaView/AgendaItem.jsx [hunk 5]
```diff
@@ -146,51 +147,100 @@ const RenderedAgendaItem = forwardRef(({ subStatus, args, ...props }, ref) => {
...
+      There was an error, please try again. This action will update the item
+      status to{" "}
+      <b>
+        <em>{t(pendingAgendaItemStatus.label)}</em>
+      </b>{" "}
+      and notify all users.`
+    </>
+  ) : (
+    <>
+      This action will update the item status to{" "}
```
Comment (JMStudiosJoe) [inline frontend/src/components/MeetingView/AgendaView/AgendaItem.jsx:236]: why the extra {" "} space?

File: frontend/src/hooks/useDeleteMeeting.jsx [hunk 6]
```diff
@@ -13,21 +24,70 @@ import DeleteMeetingModal from '../components/DeleteMeetingModal/DeleteMeetingMo
...
+    setShowModal(false);
+    history.go(0);
+  };
+
+  const openModal = () => {
+    setShowModal(true);
+  };
+
+  const [deleteMeeting, { loading, error }] = useMutation(DELETE_MEETING);
+  const date = toDateString(meeting_start_timestamp, "dddd, MMM D");
```
Comment (JMStudiosJoe) [inline frontend/src/hooks/useDeleteMeeting.jsx:44]: if we do not already have a constants file started lets go ahead and make one, add this string value there and import it in. Any new strings like this as well should also be in an object form or constant variable import for easy lookup, consistancy throughout the app, and easy to use.

**Review by JMStudiosJoe — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useDeleteMeeting.jsx (hunk 6 above)
Comment (jerrycyip) [reply frontend/src/hooks/useDeleteMeeting.jsx:44]: meeting_start_timestamp is passed in as a prop (it is a dynamic value that is different for each meeting).  I'm not clear how this is supposed to be stored as a constant value

File: frontend/src/components/MeetingView/AgendaView/AgendaItem.jsx (hunk 5 above)
Comment (jerrycyip) [reply frontend/src/components/MeetingView/AgendaView/AgendaItem.jsx:236]: I can see this is a little confusing.  I've updated this to be a template string instead so as to capture the necessary space between words.  
(Note, the modalBodyText variable is being passed into ConfirmationModal as a prop that is placed into  a paragraph tag within the confirmation modal, hence the use of an encapsulating react fragment <></>  is used here when defining modalBodyText so as to avoid un-necessary additional html tags while still passing through i18 translation logic as well as inline styling (bold and italic)).

File: frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx (hunk 2 above)
Comment (jerrycyip) [reply frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx:2]: will update

File: frontend/src/components/ConfirmationModal/ConfirmationModal.scss (hunk 4 above)
Comment (jerrycyip) [reply frontend/src/components/ConfirmationModal/ConfirmationModal.scss:7]: apologies, will fix and resubmit

File: frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx (hunk 3 above)
Comment (jerrycyip) [reply frontend/src/components/UpdateItemStartTimeModal/UpdateItemStartTimeModal.jsx:163]: noted - thanks again for mentioning this! Unfortunately my updated linter setting will update the entire jsx file to single quotes, but at least I know the expected/accepted differentiation for html tags and json would be double quotes if possible.

**Comment by jerrycyip** [comment]
Per Joey's review have applied suggested minor code updates as well as added one or two common utility functions to reduce redundancy and ensure consistency.  Code has been re-pushed and submitting for subsequent review.

**Review by JMStudiosJoe — APPROVED** [review]
